examples/recommendations/detect_and_apply_recommendations.rb

```diff
@@ -33,34 +33,34 @@ def detect_and_apply_recommendations(customer_id)
   client = Google::Ads::GoogleAds::GoogleAdsClient.new
 
   query = <<~QUERY
-    SELECT recommendation.resource_name
+    SELECT recommendation.resource_name, recommendation.campaign,
+        recommendation.keyword_recommendation
     FROM recommendation
     WHERE recommendation.type = KEYWORD
     LIMIT #{MAX_RESULT_SIZE}
   QUERY
 
   reccomendation_service = client.service.recommendation
   google_ads_service = client.service.google_ads
+
   NUMBER_OF_RUNS.times do |i|
     search_response = google_ads_service.search(
       customer_id: customer_id,
       query: query,
+      page_size: PAGE_SIZE,
     )
-    operations = search_response.map do |row|
-      client.operation.apply_recommendation do |aro|
-        aro.resource_name = row.recommendation.resource_name
-      end
-    end
 
-    unless operations.empty?
-      response = recommendation_service.apply_recommendation(
-        customer_id: customer_id,
-        operations: operations,
-      )
+    search_response.each do |row|
+      recommendation = row.recommendation
 
-      response.each do |result|
-        puts "Applied recommendation with resource name #{result.resource_name}."
+      puts "Keyword recommendation ('#{recommendation.resource_name}') was found for "\
+        "campaign '#{recommendation.campaign}'."
+      if recommendation.keyword_recommendation
+        keyword = recommendation.keyword_recommendation.keyword
+        puts "\tKeyword = '#{keyword.text}'\n\ttype = '#{keyword.match_type}'"
       end
+
+      apply_recommendation(client, customer_id, recommendation.resource_name)
     end
 
     if i < NUMBER_OF_RUNS - 1
```

Comment:
Why does this still exist if we're reducing the number of runs to 1?

Reply:
Sorry, ignore this changes!